src/nvim/api/vim.c

```diff
@@ -233,6 +233,32 @@ void nvim_set_hl(Integer ns_id, String name, Dictionary val, Error *err)
   }
 }
 
+/// Clear highlight group from a namespace
+///
+/// @param ns_id number of namespace for this highlight
+/// @param highlight_groups A list of highlight group names. These highlight
+///                         groups are cleard from namespace . If an empty
+///                         list is given entire namespace is cleared.
+void nvim_namespace_clear_hl(Integer ns_id, Array highlight_groups)
+  FUNC_API_SINCE(7)
+{
+  if (highlight_groups.size == 0) {
+    // empty name clear entire ns_id
+    ns_clear_hl((NS)ns_id);
+  } else {
+    // clear hlgroups in names from ns
+    String name = STRING_INIT;
+    for (size_t i = 0; i < highlight_groups.size; i++) {
+      if (highlight_groups.items[i].type == kObjectTypeString) {
+        name = copy_string(highlight_groups.items[i].data.string);
```

Comment:
no need to copy the string (this is probably the memory leak)

Reply:
Thanks . It worked :) You can give a review if you have time .